v-next.sql

```diff
@@ -669,3 +669,582 @@ AND NOT EXISTS (SELECT * FROM openchpl.cqm_criterion cc INNER JOIN openchpl.cqm_
 
 -- END OCD-2533
 ----------------------------------------------------
+
+-- OCD-2532 - 2015 functionality tested should be restricted by criteria
+-- 1) Create new table to support relationship between criteria and test functionality
+-- 2) Add triggers to new table
+-- 3) Rename old column
+-- 4) Add new 2015 data to new table
+-- 5) Add existing 2014 data to new table
+
+--Drop thje new table if it exists
+DROP TABLE IF EXISTS openchpl.test_functionality_criteria_map;
+
+--Create the new table
+CREATE TABLE openchpl.test_functionality_criteria_map
+(
+    id bigserial NOT NULL,
+    criteria_id bigint NOT NULL,
+    test_functionality_id bigint NOT NULL,
+    creation_date timestamp without time zone NOT NULL DEFAULT now(),
+    last_modified_date timestamp without time zone NOT NULL DEFAULT now(),
+    last_modified_user bigint NOT NULL,
+    deleted boolean NOT NULL DEFAULT false,
+    CONSTRAINT test_functionality_criteria_map_pk PRIMARY KEY (id),
+    CONSTRAINT test_functionality_criteria_fk FOREIGN KEY (criteria_id)
+        REFERENCES openchpl.certification_criterion (certification_criterion_id) MATCH FULL
+        ON UPDATE CASCADE
+        ON DELETE RESTRICT,
+    CONSTRAINT test_functionality_fk FOREIGN KEY (test_functionality_id)
+        REFERENCES openchpl.test_functionality (test_functionality_id) MATCH FULL
+        ON UPDATE CASCADE
+        ON DELETE RESTRICT
+);
+
+--Create the required triggers
+CREATE TRIGGER test_functionality_criteria_map_audit AFTER INSERT OR UPDATE OR DELETE on openchpl.test_functionality_criteria_map FOR EACH ROW EXECUTE PROCEDURE audit.if_modified_func();
+CREATE TRIGGER test_functionality_criteria_map_timestamp BEFORE UPDATE on openchpl.test_functionality_criteria_map FOR EACH ROW EXECUTE PROCEDURE openchpl.update_last_modified_date_column();
+
+-- Rename the column that will no longer be used
+DO $$
+BEGIN
+  IF NOT EXISTS(SELECT *
+                FROM information_schema.columns
+                WHERE table_catalog = 'openchpl'
+                AND table_name = 'test_functionality'
+                AND column_name = 'certification_criterion_id')
+  THEN
+      ALTER TABLE "openchpl"."test_functionality" RENAME COLUMN "certification_criterion_id" TO "certification_criterion_id";
```

Comment:
Should you be renaming it to "_deleted" here and checking for the "_deleted" column's existence in the statement above?